spec/ParseGraphQLServer.spec.js

```diff
@@ -4875,6 +4875,181 @@ describe('ParseGraphQLServer', () => {
+
+          const expectedCreateACL = {
+            __typename: 'ACL',
+            users: [
+              {
+                userId: user.id,
+                read: true,
+                write: true,
+                __typename: 'UserACL',
+              },
+              {
+                userId: user2.id,
+                read: true,
+                write: true,
+                __typename: 'UserACL',
+              },
+            ],
+            roles: [
+              {
+                roleName: 'aRole',
+                read: true,
+                write: false,
+                __typename: 'RoleACL',
+              },
+              {
+                roleName: 'aRole2',
+                read: true,
+                write: true,
+                __typename: 'RoleACL',
+              },
+            ],
+            public: { read: true, write: true, __typename: 'PublicACL' },
+          };
+
+          const query1 = new Parse.Query('SomeClass');
+          const obj1 = (await query1.get(createSomeClass.id, {
+            useMasterKey: true,
+          })).toJSON();
+
+          expect(obj1.ACL['role:aRole']).toEqual({ read: true });
+          expect(obj1.ACL['role:aRole2']).toEqual({ read: true, write: true });
+          expect(obj1.ACL[user.id]).toEqual({ read: true, write: true });
+          expect(obj1.ACL[user2.id]).toEqual({ read: true, write: true });
+          expect(obj1.ACL['*']).toEqual({ read: true, write: true });
+          expect(createSomeClass.ACL).toEqual(expectedCreateACL);
+
+          const {
+            data: { updateSomeClass },
+          } = await apolloClient.mutate({
+            mutation: gql`
+              mutation Update($id: ID!, $fields: UpdateSomeClassFieldsInput) {
+                updateSomeClass(id: $id, fields: $fields) {
+                  id
+                  ACL {
+                    users {
+                      userId
+                      read
+                      write
+                    }
+                    roles {
+                      roleName
+                      read
+                      write
+                    }
+                    public {
+                      read
+                      write
+                    }
+                  }
+                }
+              }
+            `,
+            variables: {
+              id: createSomeClass.id,
+              fields: {
+                ACL: {
+                  roles: [{ roleName: 'aRole', write: true }],
+                  public: { read: true },
+                },
+              },
+            },
+          });
+
+          const expectedUpdateACL = {
+            __typename: 'ACL',
+            users: null,
+            roles: [
+              {
+                roleName: 'aRole',
+                read: true,
+                write: true,
+                __typename: 'RoleACL',
+              },
+            ],
+            public: { read: true, write: false, __typename: 'PublicACL' },
```

Comment:
why?

Reply:
Because if you only explicitly provide `read` to Parse it seems that `write` become false.
Maybe i'm wrong ?

Reply:
Same thing here